quiz-bundle-spec.md

```diff
@@ -0,0 +1,71 @@
+
+### Items
+Items are polymorphic - i.e. there are several different _item types_ that are defined slightly differently. `items` is an array of items defined correctly for their type as defined by the `type` attribute.
+
+#### Multiple Choice Items
+
+A quiz item that has multiple options and one answer. One option is the _answer_, and the rest are _distractors_. Each option has a required rationale that explains why that option is correct or incorrect.
+
+attribute               | required | type              | notes
+----------------------- | -------- | ----------------- | -----------------------------------------
+id                      | ✓        | string            | A unique identifier for this item
+type                    | ✓        | 'multiple-choice' | The item type, which is always 'multiple-choice'
+stem                    | ✓        | dictionary        | A locale dictionary of locales for the text that asks the question, such as "Which of the following is a color?"
+options                 | ✓        | array             | An array of item options
+-- id                   | ✓        | string            | A unique ID for this item option
+-- title                | ✓        | dictionary        | A locale dictionary of locales for the content of this option, such as "Blue"
+-- rationale            | ✓        | dictionary        | A locale dictionary of locales for an explanation of why this option is correct or incorrect
+-- is_answer            | ✓        | boolean           | Whether this option is an answer or distractor
+
+#### Multiple Select Items
+
+A quiz item that has multiple options and multiple answers. Correct options are _answers_, and incorrect options are _distractors_. Each option has a required rationale that explains why that option is correct or incorrect.
+
+attribute               | required | type              | notes
+----------------------- | -------- | ----------------- | -----------------------------------------
+id                      | ✓        | string            | A unique identifier for this item
+type                    | ✓        | 'multiple-select' | The item type, which is always 'multiple-select'
+stem                    | ✓        | dictionary        | A locale dictionary of locales for the text that asks the question, such as "Which of the following is a color?"
+options                 | ✓        | array             | An array of item options
+-- id                   | ✓        | string            | A unique ID for this item option
+-- title                | ✓        | dictionary        | A locale dictionary of locales for the content of this option, such as "Blue"
+-- rationale            | ✓        | dictionary        | A locale dictionary of locales for an explanation of why this option is correct or incorrect
+-- is_answer            | ✓        | string            | Whether this option is an answer or distractor
```

Comment:
I'd like to see an explicit note on whether or not multiple correct answers are allowed (or if we deem that to be a spec violation).

I don't know what we technically support, but I'd be in favor of requiring exactly one correct answer (and not 0 or N). We can always relax to spec in the future to allow for multiple correct answers. I'd probably designate those probes as a different type so people don't accidentally pick multiple correct answers on a regular MC question.

Reply:
Yep - each MC item should have exactly 1 answer, and each MS item should have at least one answer. Will add a note for that (even if we don't validate immediately).